.github/workflows/benchmark.yml

```diff
@@ -1,50 +1,262 @@
+          echo '</details>' >> "${GITHUB_ENV}"
+
+          echo '' >> "${GITHUB_ENV}"
+
+          echo '<details>' >> "${GITHUB_ENV}"
+          echo '  <summary> Click to expand benchmark result </summary>' >> "${GITHUB_ENV}"
+          echo '' >> "${GITHUB_ENV}"
+          echo '${{ steps.read-benchmark.outputs.result }}' >> "${GITHUB_ENV}"
+          echo '' >> "${GITHUB_ENV}"
+          echo '</details>' >> "${GITHUB_ENV}"
+
+          echo 'EOF' >> "${GITHUB_ENV}"
+
+      - name: Output the comment as job summary
+        run: |
+          set -eu
+
+          echo '${{ env.PR_COMMENT }}' >> "${GITHUB_STEP_SUMMARY}"
+
+      # There is a '<!-- benchmark ci tag. exit code: {some_number} -->' comment at the beginning of the benchamrk report comment.
+      # The number in that comment is the exit code of the last benchmark run.
+      - name: Find existing comment ID
+        if: github.event_name == 'pull_request'
+        id: existing-comment
+        run: |
+          set -eu
+
+          # Known limitation: This only fetches the first 100 comments. This should not
+          # matter much because a benchmark comment should have been sent early in the PR.
+          curl -sL \
+            -X GET \
+            -H 'Accept: application/vnd.github+json' \
+            -H 'Authorization: BEARER ${{ secrets.GITHUB_TOKEN }}' \
+            -H 'X-GitHub-Api-Version: 2022-11-28' \
+            -o result.json \
+            'https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.number }}/comments?per_page=100'
+
+          # Get the last comment that has a body that starts with '<!-- benchmark ci tag. exit code: '.
+          # If available, we can just update this comment instead of creating new ones.
+          EXISTING_COMMENT=$(
+            cat result.json |
+              jq '
+                [
+                  .[] |
+                    select(.body | startswith("<!-- benchmark ci tag. exit code: "))
+                ][-1]
+              '
+          )
+
+          # Check if EXISTING_COMMENT is empty or null
+          if [ "$EXISTING_COMMENT" = "null" ] || [ -z "$EXISTING_COMMENT" ]; then
+            ID=""
+            BODY=""
+            PARSED_EXIT_CODE=""
+          else
+            ID="$(echo "$EXISTING_COMMENT" | jq '.id')"
+            BODY="$(echo "$EXISTING_COMMENT" | jq -r '.body')"
+            PARSED_EXIT_CODE="$(echo "$BODY" | head -n 1 | grep -o 'exit code: [0-9]\+' | grep -o '[0-9]\+')"
+
+            if [ "$ID" = "null" ]; then
+              echo "Comment ID was null? something is wrong"
+              echo "Comment: $EXISTING_COMMENT"
+              exit 111
+            fi
+          fi
+
+          # Output the results to GITHUB_OUTPUT
+          {
+            echo "id=$ID"
+            echo "exit-code=$PARSED_EXIT_CODE"
+            echo "body<<EOF"
+            echo "$BODY"
+            echo "EOF"
+          } >> "$GITHUB_OUTPUT"
+
+      - name: Comment in PR
```

Comment:
I think this and the previous step can be simplified, either by using the action we were using before or something like https://github.com/actions/github-script?tab=readme-ov-file#comment-on-an-issue

Reply:
That's intentional. We're only doing some simple `curl`s anyway, and i like the control and one less dependency.

Reply:
O i didn't notice you're talking about `github-scripts` even though that's the link. I knew that exists to be clear. I still like the bash more than js. Not too big of a deal though using any of them.